src/client/Client.cc

```diff
@@ -7105,7 +7105,7 @@ int Client::fill_stat(Inode *in, struct stat *st, frag_info_t *dirstat, nest_inf
     st->st_size = in->size;
     st->st_blocks = (in->size + 511) >> 9;
   }
-  st->st_blksize = MAX(in->layout.stripe_unit, 4096);
+  st->st_blksize = std::max(in->layout.stripe_unit, 4096_u32);
```

Comment:
I'm by no means a C++ guru so I'm trying to understand the utility of using `4096_u32` here. I would think the "right" way to get a u32 here is to:

`std::max<uint32_t>(...)`

So that each argument is coerced to a `uint32_t` then a comparison is done (and the compiler complains loudly if that coercion is not possible).

The version you have here wouldn't necessarily return a u32 if the first argument is e.g. a i64? (Although I believe in that case the compiler would complain about signed comparisons...)

Reply:
Either one works. Since the first argument was a uint32_t I just made the second argument be a uint32_t literal to match.

It has been suggested that that kind of literal I added is a bad idea since it doesn't catch narrowing conversions, so I might change it to the way you described.

Reply:
Is this any better than just "4096u"?  I get that it's explicitly narrowing to 32 bits but I thought the compiler could handle something like std::max((uint64)123,456u)

Reply:
I do not believe the compiler can handle that, since uint64_t and unsigned aren't the same type, so deduction fails. (Since the template is written with both arguments being the same type.) So if you do that you have to explicitly do std::max<uint64_t>, then the compiler just coerces things as usual.

This may have been the better way to go about it than adding a new literal, I just did that since I remembered someone asking about it. I can pull it out again easily.